### ISSUE TITLE
Implement method specific task queues for MethodChannels

### ISSUE THREAD

**Issue by gaaclarke** [flutter/flutter#93828]
# Description

If you have a project that is using MethodChannels and want to migrate some of the methods to execute on a background thread, not the platform thread, you have to split your usage between 2 different channels, one that handles on the platform thread and one on the background thread.  When developing federated plugins it isn't possible to change the channel communication scheme though without creating a breaking change to the platform_interface.

We have talked about removing channels from the platform_interface.  If we do that, this isn't a problem.

# Proposed Api

```java
channel = new MethodChannel("com.example.foo", binaryMessenger);
taskQueue = binaryMessenger.makeBackgroundTaskQueue();
channel.setMethodTaskQueue("foo", taskQueue);
```

**Comment by stuartmorgan** [comment]
> When developing federated plugins it isn't possible to change the channel communication scheme though without creating a breaking change to the platform_interface.

This is true only if the platform channels are part of the platform interface package; nothing about federation requires that, it's just something we did (and so is probably pretty common among 3P federated plugins as a result).

> We have talked about removing channels from the platform_interface.

For 1P plugins, which are the only ones we control.

But I don't think we probably want to create an API that adds threading subtlety just to facilitate making threading changes in a model that we will probably not recommend once we write up more docs about federate plugins.

**Comment by zanderso** [comment]
@gaaclarke Could you give a concrete example where we can see this happening?

**Comment by gaaclarke** [comment]
@zanderso here @camsim99 ran into it: https://github.com/flutter/plugins/pull/4502#discussion_r751684525

**Comment by stuartmorgan** [comment]
> We have talked about removing channels from the platform_interface. If we do that, this isn't a problem.

I just happened across this issue again while looking for something else; we have long since done this, so we no longer have a use for this feature.
